### PR TITLE
Cg 31

### PR DESCRIPTION
**Pull request by cgaddis36**
-Added status enum to item_orders table(unfulfilled fulfilled)

-Created item_orders controller to handle changing the attributes of the order status from "pending" to "cancelled", as well as changing the item_order status from "fulfilled" to "unfulfilled"

-Added in appropriating passing tests

-Added flash alert for when a user has cancelled an order

- Added functionality for changing the status of an order from pending to packaged when all item _orders have been fulfilled.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -34,6 +34,7 @@
   get "/orders/new", to: "orders#new"
   post "/orders", to: "orders#create"
   get "/orders/:id", to: "orders#show"
+  patch "/orders/:order_id", to: "item_orders#update"
```
Comment (KellyIB) [inline config/routes.rb]: I think a patch to "orders/:order_id" should be sending us to the orders controller.  I'm sorry, but could we have a conversation about this via slack?

File: spec/features/orders/user_cancel_spec.rb [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    visit '/profile/orders'
+
+    click_on(order.id)
+
+    expect(ItemOrder.last.quantity).to eq(2)
+    expect(Item.first.inventory).to eq(32)
+
+    click_on("Cancel Order")
+
+    expect(Item.first.inventory).to eq(34)
```
Comment (KellyIB) [inline spec/features/orders/user_cancel_spec.rb:143]: thank you for adding this.  you rock.

File: app/controllers/item_orders_controller.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      if itemorder.status == "fulfilled"
+        Item.find(itemorder.item_id).increment(:inventory, itemorder.quantity).save
+        itemorder.update(status: "unfulfilled")
+      end
+    end
+    order.update(status: 3)
+    flash[:alert] = "Your order has been cancelled."
+    redirect_to '/profile'
+  end
+end
```
Comment (TravisBorgsmiller) [inline app/controllers/item_orders_controller.rb:16]: I like this item_orders controller, update method looks good

File: db/migrate/20200223003017_add_status_to_item_orders.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToItemOrders < ActiveRecord::Migration[5.1]
+  def change
+    add_column :item_orders, :status, :integer, default: 0
+  end
+end
```
Comment (TravisBorgsmiller) [inline db/migrate/20200223003017_add_status_to_item_orders.rb:5]: Good idea giving the item_orders a status

**Review by TravisBorgsmiller — APPROVED** [review]

**Review by KellyIB — APPROVED** [review]

**Pull request merged**